src/Land/Model/SurfaceFlow.jl

```diff
@@ -0,0 +1,256 @@
+
+A first order flux type for the overland flow model.
+"""
+struct VolumeAdvection <: TendencyDef{Flux{FirstOrder}} end
+
+#function flux_first_order!(land::LandModel, surface::NoSurfaceFlowModel, flux::Grad, state::Vars, aux::Vars, t::Real, directions)
+#end
+
+"""
+    flux(::SurfaceWaterArea, ::VolumeAdvection, land::LandModel, args,)
+
+A first order flux method for the OverlandFlow model, adding in advection of water volume due to flowing water.
+"""
+function flux(::SurfaceWaterArea, ::VolumeAdvection, land::LandModel, args)
+    @unpack state, aux = args
+    x = aux.x
+    y = aux.y
+    width = land.surface.width(x, y)
+    area = max(eltype(state)(0.0), state.surface.area)
+    height = area / width
+    v = calculate_velocity(land.surface, x, y, height)
+    Q = area * v
+    return Q
+end
+#function flux_first_order!(land::LandModel, surface::OverlandFlowModel, fl#ux::Grad, state::Vars, aux::Vars, t::Real, directions)
+#    x = aux.x
+#    y = aux.y
+#    width = surface.width(x, y)
+#    area = max(eltype(state)(0.0), state.surface.area)
+#    height = area / width
+#    v = calculate_velocity(surface, x, y, height)
+#    Q = area * v
+#    flux.surface.area = Q
+#end 
+
+# Boundary Conditions
+
+# General case - to be used with bc::NoBC
+function surface_boundary_flux!(
+    nf,
+    bc::Land.AbstractBoundaryConditions,
+    m::Union{NoSurfaceFlowModel, OverlandFlowModel},
+    land::LandModel,
+    _...,
+) end
+
+function surface_boundary_state!(
+    nf,
+    bc::Land.AbstractBoundaryConditions,
+    m::Union{NoSurfaceFlowModel, OverlandFlowModel},
+    land::LandModel,
+    _...,
+) end
+
+"""
+    surface_boundary_flux!(
+        nf,
+        bc::Land.Dirichlet,
+        model::OverlandFlowModel,
+        land::LandModel,
+        _...,
+    )
+
+The surface boundary flux function for the OverlandFlow model, which
+does nothing if Dirichlet conditions are chosen in order to not
+overconstrain the solution.
+"""
+function surface_boundary_flux!(
+    nf,
+    bc::Land.Dirichlet,
+    model::OverlandFlowModel,
+    land::LandModel,
+    _...,
+) end
+
+"""
+    surface_boundary_state!(
+        nf,
+        bc::Land.Dirichlet,
+        model::OverlandFlowModel,
+        land::LandModel,
+        state⁺::Vars,
+        aux⁺::Vars,
+        nM,
+        state⁻::Vars,
+        aux⁻::Vars,
+        t,
+        _...,
+    )
+
+The surface boundary state function for the OverlandFlow model when
+Dirichlet conditions are chosen.
+"""
+function surface_boundary_state!(
+    nf,
+    bc::Land.Dirichlet,
+    model::OverlandFlowModel,
+    land::LandModel,
+    state⁺::Vars,
+    aux⁺::Vars,
+    nM,
+    state⁻::Vars,
+    aux⁻::Vars,
+    t,
+    _...,
+)
+    bc_function = bc.state_bc
+    state⁺.surface.area = bc_function(aux⁻, t)
+end
+
+"""
+    Precip <: TendencyDef{Source}
+
+A source term for overland flow where a prescribed net precipitation
+drives the overland flow.
+"""
+struct Precip{FT, F} <: TendencyDef{Source}
```

Comment:
I stuck this in here b/c it is only used with the SurfaceFlow model. downside is now that sources are split across multiple files..

Reply:
Yeah, that is one downside 🙁. Ultimately, it's because the types need to be defined before they are referred to.

Reply:
Hm - I actually stuck it in here because it is only used by this model, and the default for this model is to not have any prognostic variables. In the future I think it might make sense to rename `source.jl` to `soil_source.jl` unless we decide to put all sources for every component in that file. we'll see when we get more components